In [12]:
words = open("words.txt").read().splitlines()

First we read the file in a list.
Each element in the list will be a line from the file.
So each element will be a single word in this case.

In [13]:
letters = {}
for word in words:
    key = word[0]
    letters[key] = 1 if key not in letters else letters[key] + 1

We first create a dictionary which will have a letter as a key and the value will be
the amount of times it has been the first letter of a word.

For each word we take the first letter by doing word[0] then we check if the dictionary
contains it, if so we add 1 to the value otherwise we set the value to 1.

In [14]:
letters["a"]

24370

As you can see words.txt contains 24370 words that start with the letter a.

In [15]:
for key, value in letters.items():
    print(f"{key}: {value}")

2: 6
1: 8
&: 1
3: 5
4: 5
5: 3
6: 2
7: 2
8: 2
9: 2
a: 24370
A: 6480
B: 5363
b: 18969
C: 6701
c: 32232
D: 3411
d: 19367
': 33
E: 2798
e: 14017
F: 2271
f: 13599
G: 3494
g: 10972
H: 3800
h: 14862
i: 13626
y: 1245
-: 4
Y: 495
I: 1594
J: 1625
j: 2533
K: 2622
k: 3545
L: 3725
l: 10368
m: 19266
M: 5928
n: 13749
N: 2425
o: 13236
O: 1825
P: 5572
p: 35384
Q: 292
q: 2926
R: 2855
r: 18431
s: 43941
S: 6630
T: 4001
t: 21222
U: 889
u: 22902
V: 1364
v: 5452
W: 1945
w: 9727
X: 176
x: 433
Z: 638
z: 1184
